datalad/distribution/tests/test_uninstall.py

```diff
@@ -98,23 +143,147 @@ def test_uninstall_subdataset(src, dst):
         subds.repo.get(annexed_files)
 
         # uninstall data of subds:
-        res = ds.uninstall(path=subds_path, data_only=True)
+        assert_raises(ValueError, ds.uninstall, path=subds_path)
+        res = ds.uninstall(path=subds_path, recursive=True)
         ok_(all([f in res for f in annexed_files]))
         ok_(all([not i for i in subds.repo.file_has_content(annexed_files)]))
+        # subdataset is still known
+        assert_in(subds_path, ds.get_subdatasets())
 
     for subds_path in ds.get_subdatasets():
         # uninstall subds itself:
-        res = ds.uninstall(path=subds_path)
+        assert_raises(ValueError, ds.uninstall,
+                      path=subds_path, remove_handles=True, remove_history=True)
+        res = ds.uninstall(path=subds_path, remove_handles=True, remove_history=True,
+                           recursive=True)
         subds = Dataset(opj(ds.path, subds_path))
         eq_(res[0], subds)
         ok_(not subds.is_installed())
-        eq_(listdir(subds.path), [])
+        ok_(not exists(subds.path))
+
+
+@with_tree({
+    'deep': {
+        'dir': {
+            'keep': 'keep1', 'kill': 'kill1'}},
+    'keep': 'keep2',
+    'kill': 'kill2'})
+def test_uninstall_multiple_paths(path):
+    ds = Dataset(path).create(force=True)
+    # XXX with auto-save in mind it is inconsistent that `create_subdataset`
+    # doesn't save
+    subds = ds.create_subdataset('deep', force=True)
+    subds.add('.', recursive=True)
+    ds.add('.', recursive=True)
+    ds.save(auto_add_changes=True)
+    ok_clean_git(ds.path)
+    # use a tuple not a list! should also work
+    assert_raises(ValueError, ds.uninstall, ['kill', opj('deep', 'dir', 'kill')], check=False)
+    # drop content of all 'kill' files
+    ds.uninstall(('kill', opj('deep', 'dir', 'kill')), recursive=True, check=False)
+    ok_clean_git(ds.path)
+    files_left = glob(opj(ds.path, '*', '*', '*')) + glob(opj(ds.path, '*'))
+    #import pdb; pdb.set_trace()
+    ok_(all([f.endswith('keep') for f in files_left if exists(f) and not isdir(f)]))
+    ok_(all([f.endswith('kill') for f in files_left
+             if not exists(f) and not isdir(f)]))
```

Comment:
Better use `AnnexRepo.has_content` than checking for broken symlinks. This won't work in direct mode.